### PR TITLE
Fixed regression in `-alertmanager.url flag`. Basic auth was ignored.

### PR DESCRIPTION
**Pull request by Bplotka**
- Included basic auth parsing while parsing to AlertmanagerConfig
- Added test case

Fixes issue #2344 

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by brancz** [comment]
lgtm 👍 ... we probably want @fabxc to run a quick sanity check as he initially wrote this code including the backport, but thanks a lot for spotting this and fixing it so quickly!

File: cmd/prometheus/config.go [hunk 1]
```diff
@@ -365,6 +367,43 @@ func validateAlertmanagerURL(u string) error {
...
+func parseAlertmanagerURLToConfig(us string) (*config.AlertmanagerConfig, error) {
+	u, err := url.Parse(us)
+	if err != nil {
+		return nil, err
+	}
+	acfg := &config.AlertmanagerConfig{
+		Scheme:     u.Scheme,
+		PathPrefix: u.Path,
+		Timeout:    cfg.notifierTimeout,
+		ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
```
Comment (brancz) [inline cmd/prometheus/config.go]: can you just run gofmt over all of this? `make` should already do that

**Comment by fabxc** [comment]
Thanks 👍

**Comment by kfkonrad** [comment]
Is this feature still officially supported or should I consider it deprecated, now that the [YAML config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#<alertmanager_config>) seems to be stable? I like that I can specify different credentials for each Alertmanager instance, that's why I still use the flag.

**Comment by juliusv** [comment]
@kfkonrad Like with `scrape_config`, you can have multiple `alertmanager_config` sections in your configuration file, and they each allow you to specify the auth and other connection options for that group of Alertmanagers.

**Pull request merged**